lib/OpenQA/WebAPI/Controller/Admin/Influxdb.pm

```diff
@@ -90,20 +92,48 @@ sub minion ($self) {
     my $block_list = $self->app->config->{influxdb}->{ignored_failed_minion_jobs} || [];
     my $filter_jobs_num = $self->app->minion->jobs({states => ['failed'], tasks => $block_list})->total;
 
+    my $validation = $self->validation;
+    $validation->optional('rc_fail_timespan_minutes')->num;
+    return $self->reply->validation_error({format => 'json'}) if $validation->has_error;
+    my $rc_fail_timespan_minutes = $validation->param('rc_fail_timespan_minutes') // 10;
+
+    my $rc_fail_timespan_end = DateTime->now->truncate(to => 'minute');
+    # go back to last full $rc_fail_timespan_minutes minutes mark (eg from 10:37 to 10:30)
+    $rc_fail_timespan_end->subtract(minutes => ($rc_fail_timespan_end->minute() % $rc_fail_timespan_minutes));
+    # make timespan $rc_fail_timespan_minutes minutes long (eg. range from 10:20 until 10:30)
+    my $rc_fail_timespan_start = $rc_fail_timespan_end->clone()->subtract(minutes => $rc_fail_timespan_minutes);
+
+    my $dbh = $self->schema->storage->dbh;
+    # rc means hook script return code
+    my $sth = $dbh->prepare(
+        q{SELECT COUNT(*) AS rc_failed_count FROM minion_jobs
+		  WHERE finished >= ? AND finished < ? AND task = 'hook_script' AND
+		        state = 'finished' AND (notes->'hook_rc')::int != 0}
```

Comment:
Did you test how long it takes to run this query on OSD (with the time span you're going to use)? Since I was curious I've just did a test and it looks like this is definitely fast enough:

```
openqa=# EXPLAIN ANALYSE SELECT COUNT(*) AS rc_failed_count FROM minion_jobs
  WHERE finished >= '2023-06-01 15:30' AND finished < '2023-06-01 15:40' AND task = 'hook_script' AND
        state = 'finished' AND (notes->'hook_rc')::int != 0
;
                                                                                                                                   QUERY PLAN                                                                                                                                    
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Aggregate  (cost=3797.38..3797.39 rows=1 width=8) (actual time=23.938..23.939 rows=1 loops=1)
   ->  Seq Scan on minion_jobs  (cost=0.00..3797.38 rows=1 width=0) (actual time=23.929..23.929 rows=0 loops=1)
         Filter: ((finished >= '2023-06-01 15:30:00+02'::timestamp with time zone) AND (finished < '2023-06-01 15:40:00+02'::timestamp with time zone) AND (task = 'hook_script'::text) AND (state = 'finished'::minion_state) AND (((notes -> 'hook_rc'::text))::integer <> 0))
         Rows Removed by Filter: 41115
 Planning Time: 0.281 ms
 Execution Time: 24.019 ms
(6 rows
```

Reply:
yes we did some checks yesterday.